### PR TITLE
KAFKA-16878: Remove powermock and easymock from code base

### PR DESCRIPTION
**Pull request by FrankYang0529**
This is the follow-up of [KAFKA-16223](https://issues.apache.org/jira/browse/KAFKA-16223). It should include following changes:

- remove all related code from code base
- remove testsToExclude from build.gradle

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by chia7712 — APPROVED** [review]
LGTM

**Review by TaiJuWu — APPROVED** [review]

**Comment by chia7712** [comment]
@FrankYang0529 could you please rebase code to trigger QA?

**Comment by mimaison** [comment]
Wow, we're finally doing it!
According to [KAFKA-7438](https://issues.apache.org/jira/browse/KAFKA-7438), it looks like it almost took us 6 years to do it.

File: build.gradle
Comment (mimaison) [inline build.gradle:525]: There's another comment about PowerMock at line 822, can we remove it as well?

Comment (mimaison) [reply build.gradle:525]: There are also comments about EasyMock in 4 places:
- 2 in `DescribeTopicPartitionsRequestHandlerTest`
- 2 in `KafkaApisTest`
Let's clean those up as well.

**Review by mimaison — COMMENTED** [review]
Thanks for the PR!
We still have a handful of comments mentioning EasyMock/PowerMock, it would be good to remove them as well.

**Comment by chia7712** [comment]
> Wow, we're finally doing it!
According to KAFKA-7438, it looks like it almost took us 6 years to do it.

That is a lengthy journey 😄

File: build.gradle
Comment (FrankYang0529) [reply build.gradle:525]: Hi @mimaison, thanks for the review. I remove all remaining comments.

**Comment by chia7712** [comment]
@FrankYang0529 thanks for this patch. I will merge it after getting approve from @mimaison

**Comment by chia7712** [comment]
oh, the QA is not completed. I re-trigger it ...

**Comment by FrankYang0529** [comment]
> oh, the QA is not completed. I re-trigger it ...

I rebase trunk branch to include https://github.com/apache/kafka/pull/16248.

**Comment by ocadaruma** [comment]
@FrankYang0529 FYI it turned out there's another blocking test so we may need to wait https://github.com/apache/kafka/pull/16249 gets merged not only 16248

**Comment by FrankYang0529** [comment]
> @FrankYang0529 FYI it turned out there's another blocking test so we may need to wait #16249 gets merged not only 16248

Thank you for the reminder. We will wait for it 👍

**Comment by showuon** [comment]
https://github.com/apache/kafka/pull/16249 is merged. FYI.

File: build.gradle
Comment (mimaison) [inline build.gradle:829]: You can't just remove the first line, the following line does not mean anything now.

Comment (FrankYang0529) [reply build.gradle:829]: Sorry for that. I remove related comments.

**Comment by FrankYang0529** [comment]
> #16249 is merged. FYI.

Rebased it. Thanks.

**Comment by chia7712** [comment]
failed tests pass on my local.

**Comment by ijuma** [comment]
> Wow, we're finally doing it! According to [KAFKA-7438](https://issues.apache.org/jira/browse/KAFKA-7438), it looks like it almost took us 6 years to do it.

Well done team, this was a Herculean effort!

**Pull request merged**
